### ISSUE TITLE
Add support for caused_by data point

### ISSUE THREAD

**Issue by kamerynB** [project-lux/lux-frontend#309]
**Problem Description:**
The data contains a property called `caused_by` that is not being rendered.

**Expected Behavior/Solution:**
Add support for the `caused_by` data point.

**Requirements:**

- [x] Add a function in the event parser for extracting the data with its corresponding `classified_as` data point as its label
- [x] Add react component for rendering

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

- [x] Wireframe/Mockup - Kam
- [x] ~Committee discussions - Sarah~
- [x] ~Feasibility/Team discussion - Sarah~
- [x] ~Backend requirements - TBD~
- [x] Frontend requirements - Kam
- [x] Are new regression tests required for QA - Amy
- [x] ~Questions~

**UAT/LUX Examples:**

- https://lux-front-tst.collections.yale.edu/view/text/614f2504-a240-4720-8e2a-d87d9bf7eb5f

**Dependencies/Blocks:**
N/A

**Related Github Issues:**
- FE: https://github.com/project-lux/lux-frontend/issues/324

**Related links:**

- Bugherd: https://www.bugherd.com/projects/284041/tasks/2402

**Wireframe/Mockup:**
Current:
![Screenshot 2024-07-31 at 9 59 38 AM](https://github.com/user-attachments/assets/03d2b6c2-08cb-4fc3-a96b-4ac2b719b090)

Expected:
![Screenshot 2024-07-31 at 9 58 13 AM](https://github.com/user-attachments/assets/a1b2bafe-9a30-4759-9d7b-5d8cbd1b2a8f)

![image](https://github.com/user-attachments/assets/17b61ea7-df1e-4e17-9257-951c54a8f1cd)

**Comment by roamye** [comment]
- Mockup should not say contributor, it should have another label such as "Caused By". 
- The title also does not make sense for what is written in the problem/expected behavior. Title says carried_out and problem says caused_by. 
- need a better UAT example that is not from YCBA exhibitions.  
 
Needs to be updated. @kamerynB

**Comment by kamerynB** [comment]
After discussion in the 8/7 team meeting, the label should be "Caused By", not "Contributors".

**Comment by prowns** [comment]
UAT: Update labeling recommendation and mockup from 8/22 Data mtg.

**Comment by roamye** [comment]
@prowns  - 
![image](https://github.com/user-attachments/assets/04890027-cac7-45fc-a1f2-ef871f32497e)


@azaroth42 - 
![image](https://github.com/user-attachments/assets/17b61ea7-df1e-4e17-9257-951c54a8f1cd)

Was unsure of the placement so I put it under Language for now.

**Comment by prowns** [comment]
@kamerynB - do you know whhere this will be placed?

**Comment by azaroth42** [comment]
Sarah's version doesn't work because there could be additional information about people, places and times.

It would need to be:

Creation
                      Caused by Event:     Brewers
                      Additional Events:      Centential

**Comment by roamye** [comment]
@prowns & @azaroth42 - do we know which version we would want to do? If it is Sarah's I can re-update it. If not, I can just leave the screenshots above as is for now.

**Comment by prowns** [comment]
@roamye - please use Rob's version.

**Comment by kamerynB** [comment]
@azaroth42 Your example [here](https://github.com/project-lux/lux-frontend/issues/309#issuecomment-2310274811) isn't feasible with how the data is returned. What property is the frontend looking for to determine the labels? Or is it just based on the order they are parsed?

**Comment by kamerynB** [comment]
@azaroth42 As we are not using data constants anymore and to prevent the frontend from requiring an extra API call, the corresponding `/classified_as` should have a nested `equivalent` with the appropriate AAT values. In [this example](https://lux.collections.yale.edu/data/text/614f2504-a240-4720-8e2a-d87d9bf7eb5f), there are no AATs for each `/part[index]/classified_as`.

**Comment by roamye** [comment]
A data pipeline issue will need to be created as it is blocking this ticket. Will have to move it to the next milestone (not 09-09). 

Promoting as blocked and rob's mockups can replace the ones above.

**Comment by kamerynB** [comment]
Thank you all!

**Comment by roamye** [comment]
@azaroth42 @kkdavis14 - it was mentioned a data ticket will be needed as it blocks this issue. Was that ticket ever created/closed?

**Comment by kamerynB** [comment]
@roamye @prowns I'm moving this issue out of the milestone since it is blocked.

**Comment by kkdavis14** [comment]
@roamye @kamerynB this should be unblocked now, with pipeline 141 closed and implemented.

**Comment by roamye** [comment]
Approved by UAT

**Comment by roamye** [comment]
Hi, @jffcamp | @azaroth42 | @prowns 

I originally marked this as unreviewable during UAT, my mistake as it is actually reviewable through the front end. 

I spoke to @kamerynB and this is working correctly (caused_by is now a part of the JSON). However, please note that the UAT example above (https://lux.collections.yale.edu/view/text/614f2504-a240-4720-8e2a-d87d9bf7eb5f) is an instance where...
- the Editor section does not have a `Contributor` and 
- the creation section does not have `resulted from events` and `additional events`.

which is why this record does not reflect/match the mockups listed in the issue. 

There are instances where the Editor will show `Contributor` within it's section and, `resulted from events` and `additional events` within the creation section. I am still looking for UAT links, which reflect that.

 In my opinion this is fine in PROD and OK to close, but wanted to gain everyone's re-approval before closing this issue.

**Comment by kkdavis14** [comment]
@roamye @kamerynB I believe this example should have "resulted from events" and "additional events" within its Creation section, instead of "copyright" and "contributor" which are currently there. See Rob's comment from Aug 26.

**Comment by roamye** [comment]
@kamerynB - want to confirm that we need a new ticket to re-label the copyright and contributor labels and then a ticket to update the AAT labels.

**Comment by kkdavis14** [comment]
@kamerynB @roamye apologies, after looking into this further the pipeline code should be working as expected. the problem with the example record is--as perhaps you already noted--that it doesn't have the AAT equivs for the pipeline to find anything to build a data constant from. e.g. they're using their own records there:
https://linked-art.library.yale.edu/node/008df9b6-860d-4135-aac9-7780ed048094

Rob and I are looking into if there's something we can do to fix this, but perhaps there is another example out there that is working properly, and should have the "caused by Event" etc labels.

**Comment by kkdavis14** [comment]
Update: there's no YUL example where they're actually vending AATs there for pipeline to create data constants. 

Have created https://git.yale.edu/LUX/unit-data/issues/208 for YUL to vend AATs directly (best solution for this issue). In the meantime, as long as Kam's front end code is set up to accept the data constant classifications when they appear, this should be closeable.

**Comment by roamye** [comment]
@kamerynB - can you confirm Kellys comment above? 
> In the meantime, as long as Kam's front end code is set up to accept the data constant classifications when they appear, this should be closeable.

**Comment by roamye** [comment]
TF 12/05: 
- Support does not exist for pulling out those specific AAT's. 
- Would need the AAT's provided from @kkdavis14.

**Comment by kkdavis14** [comment]
Support for these AATs in creation >part > classified_as:
Creator: 300386174
Contributor: 300403974

**Comment by roamye** [comment]
Closing this as the other information looked correct in prod. I have made an issue for the support to AAT's: https://github.com/project-lux/lux-frontend/issues/483.
